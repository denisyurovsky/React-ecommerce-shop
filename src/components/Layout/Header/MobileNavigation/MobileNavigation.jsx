import PropTypes from 'prop-types';
import React, { forwardRef } from 'react';
import { useSelector } from 'react-redux';

import { USER_ROLE } from '../../../../constants/constants';
import { pathNames } from '../../../../constants/pathNames';
import { getUserRole } from '../../../../store/user/userSlice';
import { NavigationItem } from '../NavigationItem/NavigationItem';

import styles from './MobileNavigation.module.scss';

const MobileNavigation = forwardRef((props, ref) => {
  const userRole = useSelector(getUserRole);

  const navItems = [
    { title: 'HOME', path: '/' },
    { title: 'PRODUCTS', path: pathNames.PRODUCTS },
  ];

  if (userRole === USER_ROLE.ADMIN || userRole === USER_ROLE.SELLER) {
    navItems.push({ title: 'ADMIN', path: pathNames.ADMIN });
  }

  return (
    <ul
      data-testid="header-mobile-navigation"
      ref={ref}
      className={styles.container}
    >
      {navItems.map((item) => (
        <NavigationItem key={item.title} path={item.path}>
          {item.title}
        </NavigationItem>
      ))}
    </ul>
  );
});

MobileNavigation.propTypes = {
  items: PropTypes.array,
};

MobileNavigation.displayName = 'MobileNavigation';

export default MobileNavigation;
