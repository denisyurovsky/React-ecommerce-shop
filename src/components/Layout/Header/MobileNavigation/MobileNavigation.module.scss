@import "../../../../assets/styles/colors.scss";

.container {
  list-style: none;
  padding: 0;
  margin: 0;
  width: 100%;
  height: 100vh;
  position: absolute;
  left: 0;
  flex-direction: column;
  display: none;
  background-color: $primaryMain;
}
