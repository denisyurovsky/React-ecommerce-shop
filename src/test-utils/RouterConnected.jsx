import { createMemoryHistory } from 'history';
import PropTypes from 'prop-types';
import React from 'react';
import { Router, Route, Routes } from 'react-router-dom';

function RouterConnected({ component: Component, path, initialPaths }) {
  const history = createMemoryHistory({ initialEntries: initialPaths });

  return (
    <Router location={history.location} navigator={history}>
      <Routes>
        <Route path={path} element={Component} />
      </Routes>
    </Router>
  );
}

RouterConnected.propTypes = {
  component: PropTypes.object.isRequired,
  path: PropTypes.string,
  initialPaths: PropTypes.arrayOf(PropTypes.string),
};

RouterConnected.defaultProps = {
  path: '/',
  initialPaths: ['/'],
};

export default RouterConnected;
