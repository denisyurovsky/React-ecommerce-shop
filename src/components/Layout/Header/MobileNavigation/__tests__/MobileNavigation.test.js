import { render } from '@testing-library/react';
import React from 'react';

import RouterConnected from '../../../../../test-utils/RouterConnected';
import MobileNavigation from '../MobileNavigation';

describe('MobileNavigation component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const items = [
        {
          name: 'HOME',
          link: '/',
        },
        {
          name: 'PRODUCTS',
          link: '/products',
        },
      ];
      const { asFragment } = render(
        <RouterConnected component={<MobileNavigation items={items} />} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
