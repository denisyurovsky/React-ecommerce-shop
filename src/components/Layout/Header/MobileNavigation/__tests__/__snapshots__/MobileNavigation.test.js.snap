// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <ul
    class="container"
    data-testid="header-mobile-navigation"
  >
    <li
      class="item active"
      role="tab"
      tabindex="0"
    >
      <a
        href="/"
      >
        HOME
      </a>
    </li>
    <li
      class="item"
      role="tab"
      tabindex="0"
    >
      <a
        href="/products"
      >
        PRODUCTS
      </a>
    </li>
  </ul>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
